<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Running the Pipeline &#8212; Program to idenentify putative imprint control region (ICR) from Whole genome methylation data. 0.1.0 documentation</title>
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.7/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Output" href="output.html" />
    <link rel="prev" title="Installation" href="installation.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head><body>

  <div id="navbar" class="navbar navbar-inverse navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html"><span><img src="_static/my_logo.svg"></span>
          puticr</a>
        <span class="navbar-text navbar-version pull-left"><b>0.1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="installation.html">puticr</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Home <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#stable-release">Stable release</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#run-the-test-data">Run the test data</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Running the Pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#activating-the-pipeline">Activating the Pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="#to-get-help">To get help</a></li>
<li class="toctree-l2"><a class="reference internal" href="#general-pipeline-flowchart">General pipeline flowchart</a></li>
<li class="toctree-l2"><a class="reference internal" href="#usage-examples">Usage Examples</a></li>
<li class="toctree-l2"><a class="reference internal" href="#checking-error-logs">Checking error logs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html#key-output-dir-tree">Key output dir tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html#output-examples">Output Examples</a><ul>
<li class="toctree-l2"><a class="reference internal" href="output.html#cgmap-file-from-cgmaptools">CGmap file from cgmaptools</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#restrict-to-cg-context">Restrict to CG context</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#merge-symetrical-cpg-lines-updated-cgmap-file">Merge symetrical CpG lines (updated CGmap file)</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#identify-icr-hotspot-from-each-tissue-or-gamet-cells">Identify ICR hotspot from each tissue or gamet cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#merged-bed-file">Merged bed file</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#union-bed-file">Union bed file</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#count-putative-icr-in-all-tissue-final-file">Count putative ICR in all tissue (Final file)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="help.html">Help</a><ul>
<li class="toctree-l2"><a class="reference internal" href="help.html#frequently-asked-questions">Frequently Asked Questions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">puticr</a><ul>
<li class="toctree-l2"><a class="reference internal" href="puticr.html">puticr package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Credits</a><ul>
<li class="toctree-l2"><a class="reference internal" href="authors.html#development-lead">Development Lead</a></li>
<li class="toctree-l2"><a class="reference internal" href="authors.html#contributors">Contributors</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#types-of-contributions">Types of Contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#get-started">Get Started!</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#pull-request-guidelines">Pull Request Guidelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#tips">Tips</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#deploying">Deploying</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a><ul>
<li class="toctree-l2"><a class="reference internal" href="history.html#id1">0.1.0 (2018-12-18)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Running the Pipeline</a><ul>
<li><a class="reference internal" href="#activating-the-pipeline">Activating the Pipeline</a></li>
<li><a class="reference internal" href="#to-get-help">To get help</a></li>
<li><a class="reference internal" href="#general-pipeline-flowchart">General pipeline flowchart</a></li>
<li><a class="reference internal" href="#usage-examples">Usage Examples</a><ul>
<li><a class="reference internal" href="#typical-usage-with-germ-cell-tissues-such-as-kidney-liver-brain">Typical Usage with germ cell tissues such as kidney, liver, brain …</a></li>
<li><a class="reference internal" href="#gamet-cells-oocyte-and-sperm">Gamet cells (oocyte and sperm)</a></li>
<li><a class="reference internal" href="#to-index-your-own-genome-and-run-the-pipeline">To index your own genome and run the pipeline</a></li>
</ul>
</li>
<li><a class="reference internal" href="#checking-error-logs">Checking error logs</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="installation.html" title="Previous Chapter: Installation"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Installation</span>
    </a>
  </li>
  <li>
    <a href="output.html" title="Next Chapter: Output"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Output &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/analysis.rst.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12 content">
      
  <div class="section" id="running-the-pipeline">
<h1>Running the Pipeline<a class="headerlink" href="#running-the-pipeline" title="Permalink to this headline">¶</a></h1>
<div class="section" id="activating-the-pipeline">
<span id="activate"></span><h2>Activating the Pipeline<a class="headerlink" href="#activating-the-pipeline" title="Permalink to this headline">¶</a></h2>
<p>You must always ensure that the virtualenv that you installed into during the
<a class="reference internal" href="installation.html"><span class="doc">Installation</span></a> is activated prior to running any portion of the pipeline.</p>
<p>You only have to do this if you have closed your terminal since the last time your
activated.</p>
<p>You need to specify the full path to the activate script such as</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$&gt; . /path/to/puticr/puticr/bin/activate
</pre></div>
</div>
<p>You can read more about what virtualenv is <a class="reference external" href="https://virtualenv.pypa.io/en/latest/">here</a></p>
</div>
<div class="section" id="to-get-help">
<h2>To get help<a class="headerlink" href="#to-get-help" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>puticr_cli -h
</pre></div>
</div>
<p>If your fastq file has a <cite>.fq.gz</cite> extension, make sure to rename to <cite>tissueName.*.fastq.gz</cite> extension and it should be gzipped.
The name of the fastq file does matter. For example, Brain.21423B_S1_R1_001.fastq.gz, Liver.21423L_S3_R1_001.fastq.gz, oocyte.5423B_S1_R1_001.fastq.gz,
Brain.21424B_S5_R1_001.fastq.gz</p>
</div>
<div class="section" id="general-pipeline-flowchart">
<h2>General pipeline flowchart<a class="headerlink" href="#general-pipeline-flowchart" title="Permalink to this headline">¶</a></h2>
<a class="reference internal image-reference" href="_images/flowchart.png"><img alt="_images/flowchart.png" src="_images/flowchart.png" style="width: 600px;" /></a>
</div>
<div class="section" id="usage-examples">
<h2>Usage Examples<a class="headerlink" href="#usage-examples" title="Permalink to this headline">¶</a></h2>
<p>You must use  <cite>TissueName</cite>.*.fastq.gz``</p>
<div class="section" id="typical-usage-with-germ-cell-tissues-such-as-kidney-liver-brain">
<h3>Typical Usage with germ cell tissues such as kidney, liver, brain …<a class="headerlink" href="#typical-usage-with-germ-cell-tissues-such-as-kidney-liver-brain" title="Permalink to this headline">¶</a></h3>
<p>In this case, the goal is to find a region where the genome is methylated 50%. We will use the lower methylation cutoff of 0.3 and the upper cutoff 0.7.
Don’t forget to store the <cite>tissueName.*.fastq.gz</cite> file under any directory name and pass to -f option. I suggest to store the raw data into separate folders.
One for somatic tissue fastq.gz  files and the other for gamet cells (oocyte and sperm). The pipleline only  support single end reads.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>puticr_cli  -o outputDir1 -f testData/raw  -l <span class="m">6</span> -c <span class="m">12</span>  -L <span class="m">0</span>.3 -U <span class="m">0</span>.7  --indexed<span class="o">=</span>no --refGenome<span class="o">=</span>hs_chr6.fa --refGenomeDir<span class="o">=</span>testData/genome2/
</pre></div>
</div>
</div>
<div class="section" id="gamet-cells-oocyte-and-sperm">
<h3>Gamet cells (oocyte and sperm)<a class="headerlink" href="#gamet-cells-oocyte-and-sperm" title="Permalink to this headline">¶</a></h3>
<p>The goal is to find a region where the genome is methylated 100%</p>
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>puticr_cli  -o outputDir2 -f testData/raw  -l <span class="m">6</span> -c <span class="m">12</span>  -L <span class="m">0</span>.8 -U <span class="m">1</span>.0 --indexed<span class="o">=</span>no --refGenome<span class="o">=</span>hs_chr6.fa --refGenomeDir<span class="o">=</span>testData/genome2/
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="to-index-your-own-genome-and-run-the-pipeline">
<h3>To index your own genome and run the pipeline<a class="headerlink" href="#to-index-your-own-genome-and-run-the-pipeline" title="Permalink to this headline">¶</a></h3>
<p>Assuming you downlaoded and saved <cite>hs38.fa</cite> under <cite>mygenome/genome3/</cite>, you will turn on <cite>indexed=yes</cite>. Indexing a genome is a slower process, be petient.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>puticr_cli  -o outputDir1 -f testData/raw  -l <span class="m">6</span> -c <span class="m">12</span>  -L <span class="m">0</span>.3 -U <span class="m">0</span>.7  --indexed<span class="o">=</span>yes --refGenome<span class="o">=</span>hs38.fa --refGenomeDir<span class="o">=</span>mygenome/genome3/
</pre></div>
</div>
<p>Then, the next time you run a different data or the same data. You don’t need to re-index. This is a one time process unless you used different version of the genome.</p>
<p>Don’t forget to change <cite>–indexed=no</cite>.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>puticr_cli  -o outputDir2 -f testData/raw  -l <span class="m">6</span> -c <span class="m">12</span>  -L <span class="m">0</span>.8 -U <span class="m">1</span>.0 --indexed<span class="o">=</span>no --refGenome<span class="o">=</span>hs38.fa --refGenomeDir<span class="o">=</span>mygenome/genome3/
</pre></div>
</div>
</div>
</div>
<div class="section" id="checking-error-logs">
<h2>Checking error logs<a class="headerlink" href="#checking-error-logs" title="Permalink to this headline">¶</a></h2>
<p>If it fails then an error is reported that generally suggest where it failed by
checking the key files created at each stage. Most likely, the error occurs on the
suggested stage or the stage before it. You will likely have to check the log files
to get an idea what went wrong and go from there.</p>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2018 Dereje Jima.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.1.<br/>
    </p>
  </div>
</footer>
  </body>
</html>