<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Installation &#8212; Program to idenentify putative imprint control region (ICR) from Whole genome methylation data. 0.1.0 documentation</title>
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.7/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Running the Pipeline" href="analysis.html" />
    <link rel="prev" title="Putative Imprint Control Regions (ICRs) identification pipeline (puticr)" href="index.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head><body>

  <div id="navbar" class="navbar navbar-inverse navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html"><span><img src="_static/my_logo.svg"></span>
          puticr</a>
        <span class="navbar-text navbar-version pull-left"><b>0.1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="#">puticr</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Home <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#stable-release">Stable release</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-the-test-data">Run the test data</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="analysis.html">Running the Pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="analysis.html#activating-the-pipeline">Activating the Pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="analysis.html#to-get-help">To get help</a></li>
<li class="toctree-l2"><a class="reference internal" href="analysis.html#general-pipeline-flowchart">General pipeline flowchart</a></li>
<li class="toctree-l2"><a class="reference internal" href="analysis.html#usage-examples">Usage Examples</a></li>
<li class="toctree-l2"><a class="reference internal" href="analysis.html#checking-error-logs">Checking error logs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html#key-output-dir-tree">Key output dir tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html#output-examples">Output Examples</a><ul>
<li class="toctree-l2"><a class="reference internal" href="output.html#cgmap-file-from-cgmaptools">CGmap file from cgmaptools</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#restrict-to-cg-context">Restrict to CG context</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#merge-symetrical-cpg-lines-updated-cgmap-file">Merge symetrical CpG lines (updated CGmap file)</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#identify-icr-hotspot-from-each-tissue-or-gamet-cells">Identify ICR hotspot from each tissue or gamet cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#merged-bed-file">Merged bed file</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#union-bed-file">Union bed file</a></li>
<li class="toctree-l2"><a class="reference internal" href="output.html#count-putative-icr-in-all-tissue-final-file">Count putative ICR in all tissue (Final file)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="help.html">Help</a><ul>
<li class="toctree-l2"><a class="reference internal" href="help.html#frequently-asked-questions">Frequently Asked Questions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">puticr</a><ul>
<li class="toctree-l2"><a class="reference internal" href="puticr.html">puticr package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Credits</a><ul>
<li class="toctree-l2"><a class="reference internal" href="authors.html#development-lead">Development Lead</a></li>
<li class="toctree-l2"><a class="reference internal" href="authors.html#contributors">Contributors</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#types-of-contributions">Types of Contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#get-started">Get Started!</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#pull-request-guidelines">Pull Request Guidelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#tips">Tips</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#deploying">Deploying</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a><ul>
<li class="toctree-l2"><a class="reference internal" href="history.html#id1">0.1.0 (2018-12-18)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Installation</a><ul>
<li><a class="reference internal" href="#stable-release">Stable release</a></li>
<li><a class="reference internal" href="#run-the-test-data">Run the test data</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="index.html" title="Previous Chapter: Putative Imprint Control Regions (ICRs) identification pipeline (puticr)"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Putative Impr...</span>
    </a>
  </li>
  <li>
    <a href="analysis.html" title="Next Chapter: Running the Pipeline"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Running the Pipeline &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/installation.rst.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12 content">
      
  <div class="section" id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h1>
<div class="section" id="stable-release">
<h2>Stable release<a class="headerlink" href="#stable-release" title="Permalink to this headline">¶</a></h2>
<p>System dependancies:</p>
<ol class="arabic simple">
<li>Python &gt;= 2.7 &lt;- usally exist in all linux system, check the version</li>
<li>graphviz &lt;- install this as a root user</li>
</ol>
<p>To install graphviz follow package manager on your unix/linux system.</p>
<p>In Ubuntu, do this on a linux terminal as root user, if you are not super user, ask your system admin to install it.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sudo apt-get install graphviz</span>
</pre></div>
</div>
<p>In Red Hat flavor linux such as CentOs:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sudo yum install graphviz</span>
</pre></div>
</div>
<p>Note: Make sure the pip and virtualenv packages are  installed on your system.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">which pip</span>
<span class="go">which virtualenv</span>
</pre></div>
</div>
<p>If not installed,</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py</span>
<span class="go">python get-pip.py  --user</span>
</pre></div>
</div>
<p>Then, if there is no virtualenv package</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">pip install virtualenv</span>
</pre></div>
</div>
<p>To install puticr, run this command in your terminal:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">git clone https://github.com/demis001/puticr.git ; cd puticr ; make install</span>
</pre></div>
</div>
<p>This is the preferred method to install puticr, as it will always install the most recent stable release.</p>
<p>Download test data from this site <a class="reference external" href="https://www.ddjima.com/software/">https://www.ddjima.com/software/</a>  and extract the result in the same directory. The directory contains an example fastq files and indexed genome and test data.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">wget https://ddjima.com/wp-content/uploads/2019/02/testData.zip</span>
</pre></div>
</div>
<p>To extract …</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">unzip testData.zip</span>
</pre></div>
</div>
</div>
<div class="section" id="run-the-test-data">
<h2>Run the test data<a class="headerlink" href="#run-the-test-data" title="Permalink to this headline">¶</a></h2>
<p>The application has to run in Python virtual envirnoment. First,  activate the virtual environment to use the application after installation.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">source   puticr/bin/activate</span>
</pre></div>
</div>
<p>Then, run the following line to test the application …
This is also the way to run germ layer tissue samples (50% methylation)</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">puticr_cli  -o outdir2 -f testData/raw  -l 6 -c 12  -L 0.3 -U 0.7  --indexed=no --refGenome=hs_chr6.fa --refGenomeDir=testData/genome2/</span>
</pre></div>
</div>
<p>To index your own reference genome and run your own data, create  the directory to hold the reference genome, merge all chromosome and store your genome as a single fasta  file. Example, mm10.fa, hs.fa etc.</p>
<p>Then do this, <cite>indexed=yes</cite> telling the application to index before it uses it. Assumming you saved mm10.fa genome under mm10ref directory…</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">puticr_cli  -o outdir2 -f testData/raw -l 6 -c 12  -L 0.3 -U 0.7  --indexed=yes --refGenome=mm10.fa --refGenomeDir=YOURPATH/mm10ref/</span>
</pre></div>
</div>
<p>To run gamet cells samples (oocytes and sperm) that expected (100% methylation call)</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">puticr_cli  -o outdir2 -f testData/raw  -l 6 -c 12  -L 0.8 -U 1.0  --indexed=no --refGenome=hs_chr6.fa --refGenomeDir=testData/genome2/</span>
</pre></div>
</div>
<p>Note: The Germ layer tisse samples and the gamet cells samples must be called separetlly.</p>
<p>Once  you have installed the document, don’t forget to activate before each use.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">source   puticr/bin/activate</span>
</pre></div>
</div>
<p>To deactivate virtual envirnoment. Type the following command on Linux commmand line</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">deactivate</span>
</pre></div>
</div>
<p>This will exit the virtual envirnoment</p>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2018 Dereje Jima.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.1.<br/>
    </p>
  </div>
</footer>
  </body>
</html>